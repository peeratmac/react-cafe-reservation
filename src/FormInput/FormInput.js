import React, { Component } from 'react';

class FormInput extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      date: '',
      time: '',
      number: ''
    };
  }

  handleAddReservation = event => {
    event.preventDefault();
    const newReservation = {
      id: Date.now(),
      ...this.state
    };
    console.log(this.props);
    this.props.addReservation(newReservation);
    this.clearInputs();
  };

  clearInputs = () => {
    this.setState({ name: '', date: '', time: '', number: '' });
  };

  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  render() {
    return (
      <form>
        <input
          type='text'
          placeholder='Guest Name'
          name='name'
          value={this.state.name}
          onChange={event => this.handleChange(event)}
        />

        <input
          type='text'
          placeholder='Date X/XX or XX/XX'
          name='date'
          value={this.state.date}
          onChange={event => this.handleChange(event)}
        />

        <input
          type='text'
          placeholder='Time X:XX'
          name='time'
          value={this.state.time}
          onChange={event => this.handleChange(event)}
        />

        <input
          type='text'
          placeholder='Number of Guests'
          name='number'
          value={this.state.number}
          onChange={event => this.handleChange(event)}
        />

        <button onClick={event => this.handleAddReservation(event)}>
          SUBMIT
        </button>
      </form>
    );
  }
}

export default FormInput;
