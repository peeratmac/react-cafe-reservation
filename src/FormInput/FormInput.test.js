import React from 'react';
import FormInput from './FormInput';
import { shallow } from 'enzyme';

describe('FORMINPUT', () => {
  let wrapper;
  const mockAddReservation = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<FormInput addReservation={mockAddReservation} />);
  });

  it('should match FormInput Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
