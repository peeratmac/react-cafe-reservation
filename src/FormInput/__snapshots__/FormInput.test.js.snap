// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FORMINPUT should match FormInput Snapshot 1`] = `
<form>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Guest Name"
    type="text"
    value=""
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="Date X/XX or XX/XX"
    type="text"
    value=""
  />
  <input
    name="time"
    onChange={[Function]}
    placeholder="Time X:XX"
    type="text"
    value=""
  />
  <input
    name="number"
    onChange={[Function]}
    placeholder="Number of Guests"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    SUBMIT
  </button>
</form>
`;
